### ISSUE TITLE
`kaput files list` panics

### ISSUE THREAD

**Issue by mikispag** [davidchalifoux/kaput-cli#60]
```
$ RUST_BACKTRACE=full kaput files list
thread 'main' panicked at /home/miki/.cargo/registry/src/index.crates.io-6f17d22bba15001f/kaput-cli-2.4.0/src/main.rs:452:22:
fetching files: reqwest::Error { kind: Decode, source: Error("invalid value: integer `-2`, expected u32", line: 1, column: 762) }
stack backtrace:
   0:     0x58c3dbeb7b75 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h367d587903c05733
   1:     0x58c3dbee1d5b - core::fmt::write::h90d6c7093938e590
   2:     0x58c3dbeb48df - std::io::Write::write_fmt::hf166be129b5ac403
   3:     0x58c3dbeb794e - std::sys_common::backtrace::print::h16f4e54dda428f87
   4:     0x58c3dbeb8c09 - std::panicking::default_hook::{{closure}}::h421dd5e2264b9ca7
   5:     0x58c3dbeb894d - std::panicking::default_hook::hd6ec143ee36c91c5
   6:     0x58c3dbeb90a3 - std::panicking::rust_panic_with_hook::he0e1ddfdd1ef8a31
   7:     0x58c3dbeb8f84 - std::panicking::begin_panic_handler::{{closure}}::h75fca2fc3c1ec62d
   8:     0x58c3dbeb8039 - std::sys_common::backtrace::__rust_end_short_backtrace::h9d480d72ed19fe3c
   9:     0x58c3dbeb8cb7 - rust_begin_unwind
  10:     0x58c3db9445e3 - core::panicking::panic_fmt::h10c3ee83b4e55229
  11:     0x58c3db944a76 - core::result::unwrap_failed::h635713c629d81684
  12:     0x58c3db9580c7 - core::result::Result<T,E>::expect::h4b0f580dd940b78c
  13:     0x58c3db961acd - kaput::main::hd5e43af3deecb440
  14:     0x58c3db98c243 - std::sys_common::backtrace::__rust_begin_short_backtrace::h23040e52b380b035
  15:     0x58c3db984979 - std::rt::lang_start::{{closure}}::hdfeeb70c881104e9
  16:     0x58c3dbeaf080 - std::rt::lang_start_internal::h5add80e16e4e0045
  17:     0x58c3db962ad5 - main
  18:     0x70bedda53c88 - <unknown>
  19:     0x70bedda53d4c - __libc_start_main
  20:     0x58c3db944d15 - _start
  21:                0x0 - <unknown>
```

**Comment by davidchalifoux** [comment]
Hmm, I'm not sure how it's receiving a negative integer.
Can you trying sending a request directly to the Put API to see if any of the responses include a negative integer for the ID of the file or parent?

Here's a template for doing it with CURL: `curl -X GET "https://api.put.io/v2/files/list?parent_id=0" -H "Authorization: Bearer <API_TOKEN>"`

You can get your api token by calling `kaput debug`.

If you don't mind, paste the entire response but you can take out the file names.

**Comment by mikispag** [comment]
```
{"cursor":null,"files":[{"content_type":"application/x-directory","crc32":null,"created_at":"2023-11-09T23:11:05","extension":null,"file_type":"FOLDER","first_accessed_at":null,"folder_type":"REGULAR","icon":"https://api.put.io/images/file_types/folder.png","id":1260335632,"is_hidden":false,"is_mp4_available":false,"is_shared":false,"name":"REDACTED","opensubtitles_hash":null,"parent_id":0,"screenshot":null,"size":549323246,"sort_by":"DATE_ASC","updated_at":"2023-11-09T23:11:05"},{"content_type":"application/x-directory","crc32":null,"created_at":"2023-06-02T22:04:50","extension":null,"file_type":"FOLDER","first_accessed_at":null,"folder_type":"SHARED_ROOT","icon":"https://api.put.io/images/file_types/folder.png","id":-2,"is_hidden":false,"is_mp4_available":false,"is_shared":true,"name":"Items shared with you","opensubtitles_hash":null,"parent_id":0,"screenshot":null,"size":316260819021,"sort_by":"DATE_DESC","updated_at":"2023-06-02T22:04:50"}],"parent":{"content_type":"application/x-directory","crc32":null,"created_at":"2015-02-18T13:32:50","extension":null,"file_type":"FOLDER","first_accessed_at":null,"folder_type":"REGULAR","icon":"https://api.put.io/images/file_types/folder.png","id":0,"is_hidden":false,"is_mp4_available":false,"is_shared":false,"name":"Your Files","opensubtitles_hash":null,"parent_id":null,"screenshot":null,"size":6867711919,"sort_by":"DATE_DESC","updated_at":"2024-07-23T14:01:40.756096"},"status":"OK","total":2}
```

**Comment by davidchalifoux** [comment]
Interesting, I haven't tried using the file sharing feature. Looks like those are put in a special folder with a negative ID. Guess we won't be able to use unsigned ints 😢 

I'll try to push a fix for this in the next day or two, PRs are welcome though.

**Comment by davidchalifoux** [comment]
This should be fixed in v2.4.1, let me know if you're still having this issue.

**Comment by mikispag** [comment]
Thank you, it is indeed fixed. Appreciate the quick fix!

Cheers,
_Miki_
